vms/platformvm/block/builder/builder_test.go

```diff
@@ -384,6 +376,80 @@ func TestBuildBlockDropExpiredStakerTxs(t *testing.T) {
 	require.ErrorIs(tx3DropReason, txexecutor.ErrFutureStakeTime)
 }
 
+func TestBuildBlockInvalidStakingDurations(t *testing.T) {
+	require := require.New(t)
+
+	env := newEnvironment(t)
+	env.ctx.Lock.Lock()
+	defer func() {
+		require.NoError(shutdownEnvironment(env))
+		env.ctx.Lock.Unlock()
+	}()
+
+	env.config.DurangoTime = time.Time{}
+	require.True(env.config.IsDurangoActivated(env.backend.Clk.Time()))
+
+	var (
+		now                   = env.backend.Clk.Time()
```

Comment:
this should be env.state.GetTimestamp() rather than the local clock. 
It is needed just to set the end time right.